astrocut/cube_cut.py

```diff
@@ -818,10 +814,21 @@ def cube_cut(self, cube_file, coordinates, cutout_size,
             start_time = time()
 
         warnings.filterwarnings("ignore", category=wcs.FITSFixedWarning)
-        with fits.open(cube_file, mode='denywrite', memmap=True) as cube:
+        fits_options: Dict[str, Any] = {"mode": "denywrite", "lazy_load_hdus": True}
+        if cube_file.startswith("s3://"):
+            fits_options["use_fsspec"] = True
+            # block size should be:
+            # block_size <= m * hdul[1].section.shape[2] * hdul[1].section.shape[3] * 4 bytes
+            fits_options["fsspec_kwargs"] = {"default_block_size": 10_000, "anon": True}
```

Comment:
did we come to a consensus as to what the block size should actually be?

Reply:
No, there needs to be more analysis on this. I'd like to see total processing time as a function of block size to see what's optimal. Also, wouldnt the optimal block size depend on where the user is accessing our AWS server from? Isnt it that the further away they are from the server Region the slower the request will be?

Reply:
I thought we'd decided the most efficient block size would be exactly the "column" of data the user is requesting to cutout?

> Also, wouldn't the optimal block size also depend on where the user is accessing our AWS server from? Isnt it that the further away they are from the server Region the slower the request will be?

Block size efficiency should be irrespective of these things. Being further away, or having a high latency connection, would exacerbate an inefficient block size, though.  But the answer to what the most efficient block size does not vary with these parameters.

Reply:
Note: this currently hard codes to ~9.5 KB, which could be too small to be a generalized default, so this probably needs updating.

The default for s3fs is 5 MB (`5 * 2^20`), which is almost certainly too large since the entire cutouts people make can be less than that.

I wonder if this optimization is something we could do in a follow up PR? Either you or I or someone else could handle it?

We can potentially use [`FsspecMonitor()`](https://github.com/barentsen/fsspec-monitor) as verification that we are getting the optimal block size.

See this notebook for usage: https://github.com/fornax-navo/fornax-s3-subsets/blob/main/notebooks/astropy-s3-subsetting-demo.ipynb

There's also some other interesting resources in that repo including some benchmarking code (but not results).

Reply:
we spoke about this just recently, but following up here to make it more clear -- I am suggesting we make the s3fs block size configuration dynamic, not fixed, and would be calculated from the specific cutout size requested